### PR TITLE
S3 fixes

### PR DESCRIPTION
**Pull request by LorbusChris**
cc @sherine-k @vrutkovs 

Already tested, verified and deployed to smaug

### CODE REVIEW

File: base/tekton.dev/tasks/cosa-build-baseos.yaml [hunk 1]
```diff
@@ -35,7 +35,7 @@ spec:
           echo "Skipping buildfetch from S3 - bucket params not set"
         else
           cosa buildfetch \
-          --url=s3://$(params.bucket-name)/builds \
+          --url=s3://$(params.bucket-name)/$(params.bucket-name)/builds \
```
Comment (vrutkovs) [inline base/tekton.dev/tasks/cosa-build-baseos.yaml:38]: I think we need a "$(params.subfolder)" here, as we're duplicating it as bucket name

**Review by vrutkovs — CHANGES_REQUESTED** [review]
One minor nitpick

**Review by sherine-k — APPROVED** [review]

File: base/tekton.dev/tasks/cosa-build-baseos.yaml (hunk 1 above)
Comment (LorbusChris) [reply base/tekton.dev/tasks/cosa-build-baseos.yaml:38]: Yes, ideally we separate the different build streams with subdirs.
However in this case, the bucket-name would have to stay duplicated here, as that's where
`cosa buildupload s3 $(params.bucket-name)/builds` currently unergonomically uploads to (it split's the param and takes the first part as bucket-name, so we can't replace that with a subfolder param).
I'd vote for fixing this later.

**Comment by LorbusChris** [comment]
There's something slightly weird with the PLR: https://console-openshift-console.apps.smaug.na.operate-first.cloud/k8s/ns/okd-team/tekton.dev~v1beta1~PipelineRun/okd-coreos-all-4.12-pipelinerun-cpfvt

The UI says the condition was not met, yet it was (or at least should've been), and the finally tasks did in fact run, too.

File: base/tekton.dev/tasks/cosa-build-baseos.yaml (hunk 1 above)
Comment (vrutkovs) [reply base/tekton.dev/tasks/cosa-build-baseos.yaml:38]: SGTM

**Review by vrutkovs — APPROVED** [review]

**Pull request merged**